### DISCUSSION TITLE
Credit Invoice

### DISCUSSION THREAD

**Discussion by ChiefMedicalDev** [timschofield/webERP#229] [Problems / Bugs]
I have an error message when I try to credit an invoice:
Database SQL Failure : The following SQL to insert the stock movement tax detail records was used
INSERT INTO stockmovestaxes (stkmoveno, taxauthid, taxrate, taxcalculationorder, taxontax) VALUES ('431287', '11', '0.05', '1', '0')
stkmoveno is not in stjmovenos table

**Comment by dalers** [comment]
Hello @JorgeSalam . So that we can help you, please explain what code you are using.

- Did you download a zip archive? If so, what was the file name of the zip file?
- Did you clone from GitHub? If so, what respository, branch and commit are you using? If you can, please post the output from `git remote -v` and `git log -n 1`

**Reply by ChiefMedicalDev** [reply]
Hello @dalers , thank you for your response.

I downloaded the file, it was named webERP_4.15.2.zip

**Reply by dalers** [reply]
Hi JorgeSalam. Did you start with a new ("empty") database? Can you list your steps leading up to the error?

**Reply by ChiefMedicalDev** [reply]
Hi @dalers , this ERP has been working for a while, but just now I have this error message, is not often we make credit invoice.  The steps are common, just go to customers, find the customer, select Customer Transaction Inquiries and on the invoice select Credit, and when I process credit receive the error, as I explained, is not very often we make credit, I don't know how can I maintain the database or something to try.  Thanks for your help, much appreciated

**Comment by dalers** [comment]
Until someone jumps in that knows exactly what causes this error and how to fix it, you'll have to put up with me not actually helping...  ;-)

How long is "for a while"? How many total items are in the system? How many total transactions? How many credit invoices have you made? Did the last credit invoice you made work correctly? Are there any differences between the credit invoice that worked correctly and this one? E.g. how the customers are setup, how the item is setup, how the credit is setup, etc.

**Reply by ChiefMedicalDev** [reply]
Your help is very appreciated, webERP it has been used for over 6 years without any issues, there's aver a 1000 items, you can imagine how many transactions over the years, for now I'll have to find a way to make the credit invoice, thanks again for your help

**Comment by timschofield** [comment]
Hi Jorge, apologies with the delay in getting back to you on this, and
thanks to Dale for his efforts in the meantime. The error message is
talking about a table called stjmovenos. There is no table called
this, the actual constraint links to a table stockmoves. Is this a
typo, or do you have customised scripts here?

Thanks
Tim

On Tue, 12 Nov 2024 at 18:47, JorgeSalam ***@***.***> wrote:
>
> I have an error message when I try to credit an invoice:
> Database SQL Failure : The following SQL to insert the stock movement tax detail records was used
> INSERT INTO stockmovestaxes (stkmoveno, taxauthid, taxrate, taxcalculationorder, taxontax) VALUES ('431287', '11', '0.05', '1', '0')
> stkmoveno is not in stjmovenos table
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>



-- 
www.weberp.org
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by ChiefMedicalDev** [comment]
Thanks so much for your response,

I don't have customized scripts and the error message does not include that table:

Database Error : CRITICAL ERROR! NOTE DOWN THIS ERROR AND SEEK ASSISTANCE: Taxes and rates applicable to this credit note line item could not be inserted because
Cannot add or update a child row: a foreign key constraint fails (`weberp_cms`.`stockmovestaxes`, CONSTRAINT `stockmovestaxes_ibfk_1` FOREIGN KEY (`stkmoveno`) REFERENCES `stockmoves` (`stkmoveno`))
Database SQL Failure : The following SQL to insert the stock movement tax detail records was used
INSERT INTO stockmovestaxes (stkmoveno, taxauthid, taxrate, taxcalculationorder, taxontax) VALUES ('432264', '11', '0.05', '1', '0')

________________________________
I checked the DB and tables and fields included in the error are part of the DB, don't understand what the issue is

**Comment by timschofield** [comment]
Hi Jorge,

Can you look into the stockmoves table and check what the highest value of
the stkmoveno is in that table?

thanks
Tim

On Mon, 25 Nov 2024 at 19:26, JorgeSalam ***@***.***> wrote:

> Thanks so much for your response,
>
> I don't have customized scripts and the error message does not include
> that table:
>
> Database Error : CRITICAL ERROR! NOTE DOWN THIS ERROR AND SEEK ASSISTANCE:
> Taxes and rates applicable to this credit note line item could not be
> inserted because
> Cannot add or update a child row: a foreign key constraint fails
> (`weberp_cms`.`stockmovestaxes`, CONSTRAINT `stockmovestaxes_ibfk_1`
> FOREIGN KEY (`stkmoveno`) REFERENCES `stockmoves` (`stkmoveno`))
> Database SQL Failure : The following SQL to insert the stock movement tax
> detail records was used
> INSERT INTO stockmovestaxes (stkmoveno, taxauthid, taxrate,
> taxcalculationorder, taxontax) VALUES ('432264', '11', '0.05', '1', '0')
>
> ________________________________
> I checked the DB and tables and fields included in the error are part of
> the DB, don't understand what the issue is
>
>
> From: Tim Schofield ***@***.***>
> Sent: November 23, 2024 9:43 AM
> To: timschofield/webERP ***@***.***>
> Cc: Jorge Salamanca ***@***.***>; Mention ***@***.***>
> Subject: Re: [timschofield/webERP] Credit Invoice (Discussion #229)
>
> Hi Jorge, apologies with the delay in getting back to you on this, and
> thanks to Dale for his efforts in the meantime. The error message is
> talking about a table called stjmovenos. There is no table called
> this, the actual constraint links to a table stockmoves. Is this a
> typo, or do you have customised scripts here?
>
> Thanks
> Tim
>
> On Tue, 12 Nov 2024 at 18:47, JorgeSalam ***@***.***> wrote:
> >
> > I have an error message when I try to credit an invoice:
> > Database SQL Failure : The following SQL to insert the stock movement
> tax detail records was used
> > INSERT INTO stockmovestaxes (stkmoveno, taxauthid, taxrate,
> taxcalculationorder, taxontax) VALUES ('431287', '11', '0.05', '1', '0')
> > stkmoveno is not in stjmovenos table
> >
> > —
> > Reply to this email directly, view it on GitHub, or unsubscribe.
> > You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
>
>
> --
> www.weberp.org
> @TimSchofield2
> Blog: https://kwamoja.home.blog/
>
> —
> Reply to this email directly, view it on GitHub<
> https://github.com/timschofield/webERP/discussions/229#discussioncomment-11357588>,
> or unsubscribe<
> https://github.com/notifications/unsubscribe-auth/BM4ELYUSERBYR3VWKPHOZOD2CCIBVAVCNFSM6AAAAABRUXJDAKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMZVG42TQOA>.
>
> You are receiving this because you were mentioned.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/229#discussioncomment-11376401>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LFTRQZ77AA277GJCMT2CN2YDAVCNFSM6AAAAABRUXJDAKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMZXGY2DAMI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by ChiefMedicalDev** [comment]
The highest value is 390920

[cid:436ed59e-a1e9-4ef2-9e45-5cdfa75df64b]

________________________________

**Comment by timschofield** [comment]
Hi Jorge, I think this is the problem. Somehow this has got out of sync.
The SQL is trying to insert a record into stockmovestaxes with a stkmoveno
of 432264 but that record can't exist in stockmoves as it's highest number
is only 390920
What is the highest stkmoveno in the stockmoves taxes table?

Thanks
Tim

On Mon, 25 Nov 2024 at 19:50, JorgeSalam ***@***.***> wrote:

> The highest value is 390920
>
> [cid:436ed59e-a1e9-4ef2-9e45-5cdfa75df64b]
>
> ________________________________
> From: Tim Schofield ***@***.***>
> Sent: November 25, 2024 2:41 PM
> To: timschofield/webERP ***@***.***>
> Cc: Jorge Salamanca ***@***.***>; Mention ***@***.***>
> Subject: Re: [timschofield/webERP] Credit Invoice (Discussion #229)
>
>
> Hi Jorge,
>
> Can you look into the stockmoves table and check what the highest value of
> the stkmoveno is in that table?
>
> thanks
> Tim
>
> On Mon, 25 Nov 2024 at 19:26, JorgeSalam ***@***.***> wrote:
>
> > Thanks so much for your response,
> >
> > I don't have customized scripts and the error message does not include
> > that table:
> >
> > Database Error : CRITICAL ERROR! NOTE DOWN THIS ERROR AND SEEK
> ASSISTANCE:
> > Taxes and rates applicable to this credit note line item could not be
> > inserted because
> > Cannot add or update a child row: a foreign key constraint fails
> > (`weberp_cms`.`stockmovestaxes`, CONSTRAINT `stockmovestaxes_ibfk_1`
> > FOREIGN KEY (`stkmoveno`) REFERENCES `stockmoves` (`stkmoveno`))
> > Database SQL Failure : The following SQL to insert the stock movement
> tax
> > detail records was used
> > INSERT INTO stockmovestaxes (stkmoveno, taxauthid, taxrate,
> > taxcalculationorder, taxontax) VALUES ('432264', '11', '0.05', '1', '0')
> >
> > ________________________________
> > I checked the DB and tables and fields included in the error are part of
> > the DB, don't understand what the issue is
> >
> >
> > From: Tim Schofield ***@***.***>
> > Sent: November 23, 2024 9:43 AM
> > To: timschofield/webERP ***@***.***>
> > Cc: Jorge Salamanca ***@***.***>; Mention ***@***.***>
> > Subject: Re: [timschofield/webERP] Credit Invoice (Discussion #229)
> >
> > Hi Jorge, apologies with the delay in getting back to you on this, and
> > thanks to Dale for his efforts in the meantime. The error message is
> > talking about a table called stjmovenos. There is no table called
> > this, the actual constraint links to a table stockmoves. Is this a
> > typo, or do you have customised scripts here?
> >
> > Thanks
> > Tim
> >
> > On Tue, 12 Nov 2024 at 18:47, JorgeSalam ***@***.***> wrote:
> > >
> > > I have an error message when I try to credit an invoice:
> > > Database SQL Failure : The following SQL to insert the stock movement
> > tax detail records was used
> > > INSERT INTO stockmovestaxes (stkmoveno, taxauthid, taxrate,
> > taxcalculationorder, taxontax) VALUES ('431287', '11', '0.05', '1', '0')
> > > stkmoveno is not in stjmovenos table
> > >
> > > —
> > > Reply to this email directly, view it on GitHub, or unsubscribe.
> > > You are receiving this because you are subscribed to this
> thread.Message
> > ID: ***@***.***>
> >
> >
> >
> > --
> > www.weberp.org
> > @TimSchofield2
> > Blog: https://kwamoja.home.blog/
> >
> > —
> > Reply to this email directly, view it on GitHub<
> >
> https://github.com/timschofield/webERP/discussions/229#discussioncomment-11357588>,
>
> > or unsubscribe<
> >
> https://github.com/notifications/unsubscribe-auth/BM4ELYUSERBYR3VWKPHOZOD2CCIBVAVCNFSM6AAAAABRUXJDAKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMZVG42TQOA>.
>
> >
> > You are receiving this because you were mentioned.
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/timschofield/webERP/discussions/229#discussioncomment-11376401>,
>
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AAAL6LFTRQZ77AA277GJCMT2CN2YDAVCNFSM6AAAAABRUXJDAKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMZXGY2DAMI>
>
> > .
> > You are receiving this because you commented.Message ID:
> > ***@***.***>
> >
>
>
> --
> www.weberp.org <http://www.weberpafrica.com>
> @TimSchofield2
> Blog: https://kwamoja.home.blog/
>
> —
> Reply to this email directly, view it on GitHub<
> https://github.com/timschofield/webERP/discussions/229#discussioncomment-11376528>,
> or unsubscribe<
> https://github.com/notifications/unsubscribe-auth/BM4ELYT7J44OOFJJJ6BXWRL2CN4NVAVCNFSM6AAAAABRUXJDAKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMZXGY2TEOA>.
>
> You are receiving this because you were mentioned.Message ID: ***@***.***>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/229#discussioncomment-11376596>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LEG6Z2674DFRSK3EGT2CN5OTAVCNFSM6AAAAABRUXJDAKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMZXGY2TSNQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Reply by ChiefMedicalDev** [reply]
<img width="1123" height="157" alt="image" src="https://github.com/user-attachments/assets/945a5e72-f220-4fe3-a31a-07c2a1da3383" />
Hi Tim, I still have this issue, is out of sync, I forgot to follow up on this.  Right now the highest number in table stackmovestaxes field stkmoveno is 415035 and is trying to insert 453030.  I really appreciate your help with this

**Comment by ChiefMedicalDev** [comment]
The highes stkmoveno from stockmovestaxes is 395341,  still lower than the number in the error message

________________________________
